specs/opcode/3cEXTCODECOPY.md

```diff
@@ -0,0 +1,42 @@
+# EXTCODECOPY opcode
+
+## Procedure
+
+The `EXTCODECOPY` opcode pops `address`, `memory_offset`, `code_offset` and `size` from the stack. It then copies `size` bytes of code of `address` from an offset `code_offset` to the memory at the address `memory_offset`. For out-of-bound scenarios where `size > len(code) - code_offset`, EVM pads 0 to the end of the copied bytes.
+
+The gas cost of `EXTCODECOPY` opcode is dynamic and is based on the cost of memory expansion and copying (variable depending on the `size` copied to memory) and accessing the address (variable depending on whether the `address` is warm or not)
+
+## Circuit Behaviour
+
+`EXTCODECOPY` makes use of the internal execution step `CopyCodeToMemory` and loops over these steps iteratively until there are no more bytes to be copied. The `EXTCODECOPY` circuit itself only constrains the values popped from stack and call context/address read lookups.
```

Comment:
I think this explanation is incorrect, and also doesn't match the python implementation.  Currently there's no execution step called `CopyCodeToMemory`.
Instead the copy of a dynamic number of bytes is verified by the CopyCircuit, and the `EXTCODECOPY` gadget only does a lookup to the CopyTable to verify the copy of bytes.

Probably you followed the `CODECOPY` spec as reference, which is outdated as well (it doesn't mention the usage of the CopyCircuit).

Reply:
Oh thanks for your explanation! I updated both `EXTCODECOPY` and `CODECOPY` md spec in [2ba6fb9](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/257/commits/2ba6fb90ae515a6fff450b27b69cc55d1a3e24d7). 
Would you review this commit? @ed255

Reply:
Looks good!